third_party/stdlib/string.py

```diff
@@ -0,0 +1,658 @@
+ascii_letters = ascii_lowercase + ascii_uppercase
+digits = '0123456789'
+hexdigits = digits + 'abcdef' + 'ABCDEF'
+octdigits = '01234567'
+punctuation = """!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~"""
+printable = digits + letters + punctuation + whitespace
+
+# Case conversion helpers
+# Use str to convert Unicode literal in case of -U
+# TODO: use map once implemented
+# l = map(chr, xrange(256))
+l = [chr(x) for x in xrange(256)]
+_idmap = str('').join(l)
+del l
+
+# Functions which aren't available as string methods.
+
+# Capitalize the words in a string, e.g. " aBc  dEf " -> "Abc Def".
+def capwords(s, sep=None):
+    """capwords(s [,sep]) -> string
+
+    Split the argument into words using split, capitalize each
+    word using capitalize, and join the capitalized words using
+    join.  If the optional second argument sep is absent or None,
+    runs of whitespace characters are replaced by a single space
+    and leading and trailing whitespace are removed, otherwise
+    sep is used to split and join the words.
+
+    """
+    return (sep or ' ').join(x.capitalize() for x in s.split(sep))
+
+
+# Construct a translation string
+_idmapL = None
+def maketrans(fromstr, tostr):
+    """maketrans(frm, to) -> string
+
+    Return a translation table (a string of 256 bytes long)
+    suitable for use in string.translate.  The strings frm and to
+    must be of the same length.
+
+    """
+    if len(fromstr) != len(tostr):
+        raise ValueError, "maketrans arguments must have same length"
+    global _idmapL
+    if not _idmapL:
+        _idmapL = list(_idmap)
+    L = _idmapL[:]
+    fromstr = map(ord, fromstr)
+    for i in range(len(fromstr)):
+        L[fromstr[i]] = tostr[i]
+    return ''.join(L)
+
+
+
+####################################################################
+import re as _re
+
+class _multimap(object):
+    """Helper class for combining multiple mappings.
+
+    Used by .{safe_,}substitute() to combine the mapping and keyword
+    arguments.
+    """
+    def __init__(self, primary, secondary):
+        self._primary = primary
+        self._secondary = secondary
+
+    def __getitem__(self, key):
+        try:
+            return self._primary[key]
+        except KeyError:
+            return self._secondary[key]
+
+
+# class _TemplateMetaclass(type):
+#     pattern = r"""
+#     %(delim)s(?:
+#       (?P<escaped>%(delim)s) |   # Escape sequence of two delimiters
+#       (?P<named>%(id)s)      |   # delimiter and a Python identifier
+#       {(?P<braced>%(id)s)}   |   # delimiter and a braced identifier
+#       (?P<invalid>)              # Other ill-formed delimiter exprs
+#     )
+#     """
+
+#     def __init__(cls, name, bases, dct):
+#         super(_TemplateMetaclass, cls).__init__(name, bases, dct)
+#         if 'pattern' in dct:
+#             pattern = cls.pattern
+#         else:
+#             pattern = _TemplateMetaclass.pattern % {
+#                 'delim' : _re.escape(cls.delimiter),
```

Comment:
Ah, I commented that out because of Metaclass, super(...).init giving error.
Looks like re is no use there. let me know If I need to remove re from this PR.